app/controllers/spree/admin/base_controller.rb

```diff
@@ -117,6 +117,8 @@ def check_updated_tos_accepted
 
         return unless spree_user_signed_in?
 
+        return if Spree::Config.enterprises_require_tos == false
+
         return if accepted_tos?
```

Comment:
I like this version. It makes it easier to add or remove conditions. And I find it less mental load when I read:

* only for signed in users, of course, now I look only at the rest of the logic
* only for instances that have the option on, of course, let's focus on the rest
* only if the user hasn't accepted it yet, of course.

So each line I know that I don't have to keep it in mind.

But, I would have made this a helper to be called from the view without a need of a global ivar. And then it would be clearer from the beginning what the outcome should be:

```rb
def tos_need_accepting?
  return false unless spree_user_signed_in?
  return false if Spree::Config.enterprises_require_tos == false

  !accepted_tos?
end
```
:shrug: